loadbalancer/healthcheck.go

```diff
@@ -41,33 +41,36 @@ func KnownHealthChecks() string {
 	return "none, http-get, https-get, custom-http, custom-https"
 }
 
-func healthy(endpoint string, transport *http.Transport) <-chan bool {
-	statusChannel := make(chan bool)
+func createHealthCheckFnWithTimeout(healthCheckTimeout time.Duration) config.HealthCheckFn {
+	return func(endpoint string, transport *http.Transport) <-chan bool {
+		statusChannel := make(chan bool)
 
-	client := &http.Client{
-		Transport: transport,
-	}
+		client := &http.Client{
+			Transport: transport,
+			Timeout:   healthCheckTimeout,
+		}
 
-	go func() {
+		go func() {
 
-		resp, err := client.Get(endpoint)
-		if err != nil {
-			log.Warn("Error doing get on healthcheck endpoint ", endpoint, " : ", err.Error())
-			statusChannel <- false
-			return
-		}
+			resp, err := client.Get(endpoint)
+			if err != nil {
+				log.Warn("Error doing get on healthcheck endpoint ", endpoint, " : ", err.Error())
+				statusChannel <- false
+				return
```

Comment:
if the error is a timeout reading the response you shoud read all the resp body before return:
https://play.golang.org/p/z9Som2cs6X